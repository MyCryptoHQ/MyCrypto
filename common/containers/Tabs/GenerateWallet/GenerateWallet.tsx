import React, { Component } from 'react';
import Keystore from './components/Keystore';
import Mnemonic from './components/Mnemonic';
import WalletTypes from './components/WalletTypes';
import CryptoWarning from './components/CryptoWarning';
import TabSection from 'containers/TabSection';
import { RouteComponentProps } from 'react-router-dom';

export enum WalletType {
  Keystore = 'keystore',
  Mnemonic = 'mnemonic'
}

export default class GenerateWallet extends Component<RouteComponentProps<{}>> {
  public render() {
    const walletType = this.props.location.pathname.split('/')[2];
    let content;

    if (window.crypto) {
      if (walletType === WalletType.Mnemonic) {
        content = <Mnemonic />;
      } else if (walletType === WalletType.Keystore) {
        content = <Keystore />;
      } else {
        content = <WalletTypes />;
      }
    } else {
      content = <CryptoWarning />;
    }

    return (
      <TabSection>
        <section className="Tab-content">{content}</section>
      </TabSection>
    );
  }
}
