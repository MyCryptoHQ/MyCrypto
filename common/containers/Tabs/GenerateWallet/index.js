// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as generateWalletActions from 'actions/generateWallet';
import type {
  GenerateNewWalletAction,
  ContinueToPaperAction,
  ResetGenerateWalletAction
} from 'actions/generateWallet';

import EnterPassword from './components/EnterPassword';
import DownloadWallet from './components/DownloadWallet';
import PaperWallet from './components/PaperWallet';
import type PrivKeyWallet from 'libs/wallet/privkey';
import type { State } from 'reducers';

type Props = {
  // Redux state
  activeStep: string, // FIXME union actual steps
  password: string,
  wallet: ?PrivKeyWallet,
  walletPasswordForm: Object,
  // Actions
  generateNewWallet: (pw: string) => GenerateNewWalletAction,
  continueToPaper: () => ContinueToPaperAction,
  resetGenerateWallet: () => ResetGenerateWalletAction
};

class GenerateWallet extends Component {
  props: Props;

  componentWillUnmount() {
    this.props.resetGenerateWallet();
  }

  render() {
    const { activeStep, wallet, password } = this.props;
    let content;

    switch (activeStep) {
      case 'password':
        content = (
          <EnterPassword
            walletPasswordForm={this.props.walletPasswordForm}
            generateNewWallet={this.props.generateNewWallet}
          />
        );
        break;

      case 'download':
        if (wallet) {
          content = (
            <DownloadWallet
              wallet={wallet}
              password={password}
              continueToPaper={this.props.continueToPaper}
            />
          );
        }
        break;

      case 'paper':
        if (wallet) {
          content = <PaperWallet wallet={wallet} />;
        } else {
          content = <h1>Uh oh. Not sure how you got here.</h1>;
        }
        break;

      default:
        content = <h1>Uh oh. Not sure how you got here.</h1>;
    }

    return (
      <section className="Tab-content">
        {content}
      </section>
    );
  }
}

function mapStateToProps(state: State) {
  return {
    walletPasswordForm: state.form.walletPasswordForm,
    activeStep: state.generateWallet.activeStep,
    password: state.generateWallet.password,
    wallet: state.generateWallet.wallet
  };
}

export default connect(mapStateToProps, generateWalletActions)(GenerateWallet);
