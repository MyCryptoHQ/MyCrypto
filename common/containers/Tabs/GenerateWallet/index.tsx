import {
  continueToPaper,
  generateNewWallet,
  resetGenerateWallet,
  TContinueToPaper,
  TGenerateNewWallet,
  TResetGenerateWallet
} from 'actions/generateWallet';
import { IFullWallet } from 'ethereumjs-wallet';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { AppState } from 'reducers';
import DownloadWallet from './components/DownloadWallet';
import EnterPassword from './components/EnterPassword';
import PaperWallet from './components/PaperWallet';
import CryptoWarning from './components/CryptoWarning';
import TabSection from 'containers/TabSection';

interface Props {
  // Redux state
  activeStep: string; // FIXME union actual steps
  password: string;
  wallet: IFullWallet | null | undefined;
  // Actions
  generateNewWallet: TGenerateNewWallet;
  continueToPaper: TContinueToPaper;
  resetGenerateWallet: TResetGenerateWallet;
}

class GenerateWallet extends Component<Props, {}> {
  public componentWillUnmount() {
    this.props.resetGenerateWallet();
  }

  public render() {
    const { activeStep, wallet, password } = this.props;
    let content;

    const AnyEnterPassword = EnterPassword as new () => any;

    if (window.crypto) {
      switch (activeStep) {
        case 'password':
          content = <AnyEnterPassword generateNewWallet={this.props.generateNewWallet} />;
          break;

        case 'download':
          if (wallet) {
            content = (
              <DownloadWallet
                wallet={wallet}
                password={password}
                continueToPaper={this.props.continueToPaper}
              />
            );
          }
          break;

        case 'paper':
          if (wallet) {
            content = <PaperWallet wallet={wallet} />;
          } else {
            content = <h1>Uh oh. Not sure how you got here.</h1>;
          }
          break;

        default:
          content = <h1>Uh oh. Not sure how you got here.</h1>;
      }
    } else {
      content = <CryptoWarning />;
    }

    return (
      <TabSection>
        <section className="Tab-content">{content}</section>
      </TabSection>
    );
  }
}

function mapStateToProps(state: AppState) {
  return {
    activeStep: state.generateWallet.activeStep,
    password: state.generateWallet.password,
    wallet: state.generateWallet.wallet
  };
}

export default connect(mapStateToProps, {
  generateNewWallet,
  continueToPaper,
  resetGenerateWallet
})(GenerateWallet);
