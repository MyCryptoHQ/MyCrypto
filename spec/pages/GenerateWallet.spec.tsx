import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import GenerateWallet from 'containers/Tabs/GenerateWallet';
import shallowWithStore from '../utils/shallowWithStore';
import { createMockStore } from 'redux-test-utils';

Enzyme.configure({ adapter: new Adapter() });

it('render snapshot', () => {
  const testState = {
    form: {
      walletPasswordForm: {}
    },
    generateWallet: {
      activeStep: {},
      password: {},
      wallet: {}
    }
  };
  const store = createMockStore(testState);
  const component = shallowWithStore(<GenerateWallet />, store);

  expect(component).toMatchSnapshot();
});
