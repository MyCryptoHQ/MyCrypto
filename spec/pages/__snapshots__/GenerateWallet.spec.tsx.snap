// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<GenerateWallet
  activeStep={Object {}}
  continueToPaper={[Function]}
  generateNewWallet={[Function]}
  password={Object {}}
  resetGenerateWallet={[Function]}
  wallet={Object {}}
  walletPasswordForm={Object {}}
/>
`;
